src/zodbbrowser/references.py

```diff
@@ -0,0 +1,136 @@
+
+import sqlite3
+
+from ZODB.utils import u64
+from ZODB.serialize import referencesf
+from zope.interface import implements
+
+from zodbbrowser.interfaces import IReferencesDatabase
+
+
+def connect(callback):
+    """Decorator for the reference database to access the sqlite DB.
+    """
+
+    def wrapper(self, *args, **kwargs):
+        try:
+            connection = sqlite3.connect(self.db_name)
+        except:
+            raise ValueError('impossible to open references database')
+        try:
+            result = callback(self, connection, *args, **kwargs)
+        except:
+            connection.close()
+            raise
```

Comment:
You're not closing the connection if the callback doesn't raise an error.  Why not?